### PR TITLE
DonghaoYang_45930032_Report_Problem2_GCN

### PR DESCRIPTION
**Pull request by ElonQuasimodoYoung**
This pull request is the first pull request of my GCN classification algorithm. The dataset.py is the script for preprocessing the data. The modules.py is the script for building the GCN model. The train.py is the script for training, validating and testing and saving the GCN model. The predict.py is the script for loading and using the pretrained GCN model. The readMeImage folder includes the images used by the README.md file. The 4_layers_GCN_model.h5 is a saved trained GCN model. The GCN_model_training_and_validation_accuracy.png is the train and validation accuracy plot. The GCN_model_training_and_validation_loss.png is the train and validation loss plot. The predict_output is the output of running the predict.py. The train_output is the output of running the train.py. The TSNE_embedding_plot.png is the TSNE plot produced by the train.py.

### CODE REVIEW

**Comment by gayanku** [comment]
GCN code, UMAP and results are OK

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove DS_Store and .idea files -2
Request Description good

**Comment by ElonQuasimodoYoung** [comment]
> ## Good Practice (Design/Commenting, TF/Torch Usage)
> Adequate use and implementation Good spacing and comments Header blocks
> 
> ## Recognition Problem
> Solves problem Driver Script present File structure present Shows Usage & Demo & Visualisation & Data usage Module present Commenting No Data leakage Difficulty: Normal -5
> 
> ## Commit Log
> Meaningful commit messages Progressive commits used
> 
> ## Documentation
> ReadMe good Good Description and Comments Markdown used PDF submitted
> 
> ## Pull Request
> Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) Feedback required, remove DS_Store and .idea files -2 Request Description good

Hi, I deleted all .DS_Store files. However, I didn't find any .idea files in my repository.
